App 정보 최신화 및 통합 테스트

doitduri opened the pull request:
### 작업 내용
- [x] 앱 아이콘 교체
- [x] 폴더 구조 변경
- [x] `HomeViewModel` 추가
- [x] 기존 LinkButton > outlink로 변경
- [x] 지도, 캘린더 기능 추가
- [x] iPad TabView 변경 건 대응
- [x] API 변경 건 및 Model 오류 수정  

---
### Summary

|항목|내용|비고|
|------|---|---|
|앱 아이콘 교체|![스크린샷 2024-11-12 오후 9 58 17](https://github.com/user-attachments/assets/5cda49be-884c-4f40-b53e-e80e0aabeac2)|디자인 변경 건 적용|
|폴더 구조 변경| Views/ViewModels/Models 로 그룹핑 | AS-IS : Views 하위 ViewModel, Model 존재 |
|HomeViewModel 추가| 캘린더 권한 관리, map UrlScheme data 관리 | |
|기존 LinkButton > outlink로 변경 | 더보기 탭과 동일하게 outlink(사파리)로 변경  | AS-IS WebView 사용 |
|지도, 캘린더 기능 추가 | 지도 선택 시, 애플 지도로 이동 / 캘린더 선택 시 권한 확인 및 일정 추가  | |
|iPad TabView 변경 건 대응 | iOS 18 TabView 신규 화면 미대응으로 변경 | 타임 테이블, 이전 행사 해당 |
|API 변경 건 및 Model 오류 수정 | speaker 정보 변경 건, VideoItem identifier 수정  |  |

doitduri committed:
fix: 앱 아이콘 교체

doitduri committed:
fix: 행사 정보 최신화

[변경 점]
- 티켓 판매 홈페이지(festa) url
- 뉴스레터 url
- 행사 시간 정보

doitduri committed:
remove: 폴더 구조 변경 및 정리

- Views / ViewModels / Models 구조로 변경
- 기타 공통으로 사용되는 View 계층 > Common으로 이동

doitduri committed:
refactor: Home 미사용 코드 제거 및 outlink로 변경

doitduri committed:
refactor: Home Link 로직 ViewModel로 이관

doitduri committed:
add: 캘린더 저장 로직 추가

doitduri committed:
add: 지도 선택 시 지도 앱 불러오기 추가

doitduri committed:
remove: 미사용 코드 삭제(WebView)

doitduri committed:
fix: VideoItem identifiable key 값 교체

doitduri committed:
fix: API > Session Model 변경 건 적용

doitduri committed:
fix: organizerURL endPoint 업데이트

doitduri committed:
fix: iPad SplitView 미대응으로 변경 (타임 테이블, 이전 행사)

doitduri committed:
fix: iOS 18 TabView 일 경우, tint 컬러 우회 로직 추가

JINi0S reviewed (approved):
제 엑스코드&기기가 최근 버전이 아니어서,,,!!! 어제 디스코드로 보내드린 부분&관련된 부분 주석처리하고 빌드해서 테스트해봤는데,

제 기기에서는 지난 행사 탭에서 목록 선택했을 때 유튜브로 넘어가질 않네요 🥲‼️
버전문제인걸까요?!!!! 💣💣💣💣💣

doitduri committed:
fix: 아이패드 세로 고정으로 변경

doitduri committed:
fix: iPad iOS 18 대응 오류 수정

- iPad 지난 행사 > 선택 시 상세 페이지로 이동되는 현상 수정
- TabBar, NaviagitonController tint 값 적용

doitduri committed:
fix: 스탬프 이벤트 정책 변경 건 적용

doitduri commented:
> 제 엑스코드&기기가 최근 버전이 아니어서,,,!!! 어제 디스코드로 보내드린 부분&관련된 부분 주석처리하고 빌드해서 테스트해봤는데,
> 
> 제 기기에서는 지난 행사 탭에서 목록 선택했을 때 유튜브로 넘어가질 않네요 🥲‼️ 버전문제인걸까요?!!!! 💣💣💣💣💣

아이코 . ㅠ제가 코드 빼먹은 부분이 있었네요..! 수정 했습니다 :)

doitduri committed:
remove: 미사용코드 삭제

doitduri committed:
add: xcprivacy 추가

doitduri committed:
fix: 문구 수정(페스타, 타임테이블)

doitduri committed:
fix: 이름 노출 정책 적용

doitduri committed:
fix: iPad > youtube 재생 후 화면 이탈 후에도 사운드가 재생되는 현상

doitduri committed:
fix: 빌드 버전 정책 적용(마이너 버전 표기)
